rust: Clarify bb-2 exercises per feedback (#218)
pingcap/talent-plan: rust/building-blocks/bb-2.md

  know the behavior of a variety of data structures well, if not their
  implementations. The standard library's `collections` module has a pretty
  amazing overview of the tradeoffs between a number of the most common
  collection types in computer science. For this, you only need to read the
  module docs.

- **[Reading: `std::io`][io]**. Again, you need to know your I/O tools, and
  again the Rust `io` module docs, while not as brilliant reading as the
  `collections` docs, provide a good overview of your toolset. You only need to
  read the module docs.

- **Exercise: Serialize and deserialize a data structure with `serde` (JSON)**.

  This exercise and the next two will introduce basic serialization and
  deserialization with [`serde`]. `serde` serializes data quickly and is easy to
  use, while also being extensible and expressive.

  For your serializable data structure, imagine a flat game-playing surface
  covered in a grid of squares, like a chess board. Imagine you have a game
  character that every turn may move any number of squares in a single
  direction. Define a type, `Move` that represents a single move of that

>   charecter.

Reviewer: ```suggestion
  character.
```

  Derive the [`Debug`] trait so `Move` is easily printable with the `{:?}`
  format specifier.

  Write a `main` function that defines a variable, `a`, of type `Move`,
  serializes it with [`serde`] to a [`File`], then deserializes it back again to a
  variable, `b`, also of type `Move`.

  Use [JSON] as the serialization format.

  Print `a` and `b` with `println!` and the `{:?}` format specifier to verify
  successful deserialization.

  Note that the `serde` book has many [examples] to work off of.

- **Exercise: Serialize and deserialize a data structure to a buffer with
    `serde` (RON)**.

  Do the same as above, except this time, instead of serializing to a `File`,

>   serialize to a `Vec<u8>` buffer, and instead of JSON as the format, use [RON].

Reviewer: ```suggestion
  serialize to a `Vec<u8>` buffer, after that try using [RON] instead of JSON as the format.
```
